web/src/components/common/FormRow.tsx

```diff
@@ -81,28 +87,105 @@ const ErrorContainer = styled.div`
   color: var(--bright-red);
 `;
 
+interface InputProps {
+  inputWidth?: string;
+  inputStyle: FormInputStyle;
+}
+
+const flatInputFieldStyle = css`
+  background-color: var(--pale-gray);
+  border-radius: 10px;
+  color: var(--dark-gray);
+
+  width: 100%;
+  box-sizing: border-box;
+
+  padding: 0.8em 1em;
+`;
+
+const shadowInputFieldStyle = css`
+  border-radius: 15px;
+  box-shadow: 4px 2px 4px rgba(0, 0, 0, 0.25);
+
+  padding: 0.6em 1em;
+`;
+
+const inputFieldStyle = css`
+  border: none;
+  font-size: 1em;
+
+  width: ${({inputWidth}: InputProps) => inputWidth || '100%'};
+
+  /* stylelint-disable value-keyword-case */
+  ${({inputStyle}: InputProps) => {
+    switch (inputStyle) {
+      case 'shadow':
+        return shadowInputFieldStyle;
+      case 'flat':
+      default:
+        return flatInputFieldStyle;
+    }
+  }};
+  /* stylelint-enable value-keyword-case */
+`;
+
+const Input = styled.input`
+  ${inputFieldStyle}/* stylelint-disable-line value-keyword-case */
+`;
+
+const TextArea = styled.textarea`
+  ${inputFieldStyle}/* stylelint-disable-line value-keyword-case */
+`;
+
 interface FormRowProps {
   index: number;
+  textAreaRows?: number;
+  inputWidth?: string;
+  stacked?: boolean;
+  inputStyle?: FormInputStyle;
 }
 
 /**
- * This is a row in a form, consisting of a label and an input field shown
- * side by side. This also contains a container for error message which is
+ * FormRow is a row in a form, consisting of a label and an input field, styled according
+ * to inputStyle. FormRow also contains a container for error message which is
  * displayed below the input field where applicable.
+ * Index is the index of the row in the form.
```

Comment:
pls use @param

Reply:
@ifhuang At first I was reluctant to use `@param` because they are props so `@param index <desc>` is does not really make sense.
But I guess I can do  `@param props.index <desc>`